packages/api-server/src/methods/modules/eth.ts

```diff
@@ -859,6 +861,72 @@ async function allTypeEthAddressToAccountId(
   return accountId;
 }
 
+async function ethCallTx(
+  txCallObj: TransactionCallObject,
+  rpc: GodwokenClient,
+  isEthWallet: boolean,
+  blockNumber?: U64
+): Promise<RunResult> {
+  const toAddress = txCallObj.to || "0x" + "00".repeat(20);
+
+  // if eth wallet mode, and `toAddress` not in allow list, reject.
+  if (isEthWallet && !allowedAddresses.has(toAddress.toLowerCase())) {
+    throw new Web3Error("not supported to address!");
+  }
+
+  const abi = new Abi(SUDT_ERC20_PROXY_ABI);
+  const ethToGwAddr = async (addr: HexString): Promise<HexString> => {
+    const result = await allTypeEthAddressToShortAddress(rpc, addr);
+    return result!;
+  };
+
+  // TODO: find by db.addresses when not found
+  const gwToEthAddr = async (addr: HexString): Promise<HexString> => {
+    const scriptHash = await rpc.getScriptHashByShortAddress(addr);
+    if (scriptHash == null) {
+      // return undefined;
+      throw new Web3Error(`eth address by short address ${addr} not found!`);
+    }
+    const script = await rpc.getScript(scriptHash);
+    if (script == null) {
+      // return undefined;
+      throw new Web3Error(`eth address by short address ${addr} not found!`);
+    }
+    return "0x" + script.args.slice(66, 106);
+  };
+
+  const data: HexString | undefined = txCallObj.data || "0x0";
+  if (isEthWallet) {
+    const dataWithShortAddress = await abi.refactor_data_with_short_address(
+      data,
+      ethToGwAddr
+    );
+    // replace data
+    txCallObj.data = dataWithShortAddress;
+  }
+
+  const rawL2Transaction = await buildEthCallTx(txCallObj, rpc);
+  const runResult = await rpc.executeRawL2Transaction(
+    rawL2Transaction,
+    blockNumber
+  );
+
+  const abiItem = abi.get_intereted_abi_item_by_encoded_data(data);
+
+  if (isEthWallet) {
```

Comment:
```suggestion
  if ( abiItem && isEthWallet) {
```
I think abiItem here might be empty